downsample.py

```diff
@@ -0,0 +1,163 @@
+"""
+Import ZDF point cloud and downsample it.
+"""
+
+from math import fmod
+import numpy as np
+import matplotlib.pyplot as plt
+import zivid
+from vtk_visualizer import plotxyzrgb
+
+
+def gridsum(matrix, downsampling_factor):
+    """
+    Reshape and sum in second direction
+    """
+    return sumline(
+        np.transpose(sumline(matrix, downsampling_factor)), downsampling_factor
+    )
+
+
+def sumline(matrix, downsampling_factor):
+    """
+    Reshape and sum in first direction
+    """
+    return np.transpose(
+        np.nansum(
+            np.transpose(np.transpose(matrix).reshape(-1, downsampling_factor)), 0
+        ).reshape(
+            int(np.shape(np.transpose(matrix))[0]),
+            int(np.shape(np.transpose(matrix))[1] / downsampling_factor),
+        )
+    )
+
+
+def downsample(xyz, rgb, contrast, downsampling_factor):
```

Comment:
perhaps add an underscore (_) for all the functions indicating that these are private functions, and not something that should be imported. Same for the other non-underscore functions.

Reply:
done